src/com/amazon/ion/impl/IonCursorBinary.java

```diff
@@ -1345,11 +1348,6 @@ private boolean slowReadValueHeader(IonTypeID valueTid, boolean isAnnotated, Mar
         if (valueTid.isDelimited) {
             endIndex = DELIMITED_MARKER;
         } else if (valueTid.variableLength) {
-            // At this point the value must be at least 2 more bytes: 1 for the smallest-possible value length
-            // and 1 for the smallest-possible value representation.
-            if (!fillAt(peekIndex, 2)) {
-                return true;
-            }
```

Comment:
The changes in this file are to support encodings like `F5 01`, a variable-length integer with a declared length of 0. Previously it was assumed that variable-length types would always have length at least 1, because there is no good reason to encode zero-length values with anything other than the dedicated opcode.

Reply:
Would tagged-type macro parameters be a good reason?

Reply:
Would you need to use a variable-length opcode for those, as opposed to a zero-length opcode?

Reply:
Maybe there's a macro that accepts an arbitrarily large tagged integer, but when you invoke it, you want to pass in `0`.

Reply:
Unless I'm missing something, you'd still use the dedicated zero opcode in that case.